### PR TITLE
sql: add some more verbose job descriptions

### LINKED ISSUE #46769: sql: improve the schema change job descriptions that aren't tied to statements
In some cases, during schema changes, we start schema change jobs for table descriptor updates that are auxiliary to the main schema change job, such as updating foreign key backreferences, or that don't have a statement tied to them for some reason. These jobs currently have very vague descriptions which may be confusing or unhelpful. See #46757, where the job description in this case is just `updating privileges`. These strings could at least be updated with table IDs, etc.

### PR DESCRIPTION
**Pull request by jordanlewis**
Closes #46769.

Release note: None

### CODE REVIEW

File: pkg/sql/colexec/types.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 The Cockroach Authors.
```
Comment (spaskob) [inline pkg/sql/colexec/types.go:1]: should this file be part of this commit?

File: pkg/sql/logictest/testdata/logic_test/notify [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# LogicTest: local
```
Comment (spaskob) [inline pkg/sql/logictest/testdata/logic_test/notify:1]: should this file be part of this commit?

**Review by spaskob — APPROVED** [review]
This is very handy!
Approving modulo the TODO comment to Lucy that asks about empty job desvription.

**Review by thoszhang — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50350)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @jordanlewis, @lucy-zhang, and @spaskob)

___
*[pkg/sql/alter_table.go, line 839 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50350#-MA2737O-TIBTM-Mvej2:-MA2737O-TIBTM-Mvej3:b-jq38ha) ([raw file](https://github.com/cockroachdb/cockroach/blob/6ce87e04e62e63fa707559d1861991ac0fd208f1/pkg/sql/alter_table.go#L839)):*
> ```Go
> 			}
> 			for _, changedSeqDesc := range changedSeqDescs {
> 				// TODO (lucy): Have more consistent/informative names for dependent jobs.
> ```

I think these TODOs can go away.
___
*[pkg/sql/create_table.go, line 1033 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/50350#-MA279zs-OSZ6BuBuRuG:-MA279zs-OSZ6BuBuRuH:bb073w6) ([raw file](https://github.com/cockroachdb/cockroach/blob/6ce87e04e62e63fa707559d1861991ac0fd208f1/pkg/sql/create_table.go#L1033)):*
> ```Go
> 
> 		// No job description, since this is presumably part of some larger schema change.
> 		// TODO(lucy): what is the right job description here?
> ```

That's a good question. My original assumption was that this should always be an empty string, since it gets appended to some other job description for a `CREATE TABLE`/`CREATE INDEX` statement (which is what happens when `writeSchemaChange` gets called multiple times on the same table descriptor in the same transaction). But I suppose it wouldn't hurt for it to be `setting metadata for interleaved table` or something.

**Comment by jordanlewis** [comment]
TFTRs. I removed the accidentally added files. I'll leave the empty job descs alone for now because your justification seems fine...

**Pull request merged**